keras/layers/core/einsum_dense.py

```diff
@@ -335,23 +345,49 @@ def quantized_build(self, input_shape, mode):
             )
 
     def quantized_call(self, inputs):
-        if self.lora_enabled:
-            raise ValueError("`quantized_call` doesn't support lora weights")
-        inputs, inputs_scale = self.inputs_quantizer(inputs)
-        x = ops.einsum(self.equation, inputs, self.kernel)
-        # Deal with `inputs_scale`
-        inputs_scale = ops.transpose(inputs_scale, self._input_transpose_axes)
-        if self._input_expand_axes:
-            inputs_scale = ops.expand_dims(
-                inputs_scale, axis=self._input_expand_axes
-            )
-        if self._input_squeeze_axes:
-            inputs_scale = ops.squeeze(
-                inputs_scale, axis=self._input_squeeze_axes
+        @ops.custom_gradient
+        def einsum_with_inputs_gradient(inputs, kernel, kernel_scale):
+            def grad_fn(*args, upstream=None):
+                if upstream is None:
+                    (upstream,) = args
+                float_kernel = ops.divide(
+                    ops.cast(kernel, dtype=self.compute_dtype),
+                    kernel_scale,
+                )
+                # From https://stackoverflow.com/a/47609896
```

Comment:
Wow. Brilliant! Never thought of it this way!! A very beautiful way to exploit Einsteinian Tensor Summation.